lib/diagnostic.js

```diff
@@ -38,19 +44,36 @@ let owner = null;
 // Methods: vocalize
 
 // Replace null with your answer
-let pet = null;
+let pet = function(name, species, breed, vocalization) {};
 
+pet.prototype.vocalize = function() { };
+console.log(pet.prototype);
 // 4. Create a constructor function for pet objects, as described above. The
 // constructor function should be responsible for setting the values of `name`,
 // `species`, `breed`, and `vocalization`.
 
-const Pet = function () {
-  /* Your answer here */
+const Pet = function (name,species,breed,vocalization) {
+  this.name = name;
+  this.species = species;
+  this.breed = breed;
+  this.vocalization = vocalization;
 };
 
 // 5. Attach a `vocalize` method to the Pet's prototype.
-
-/* your answer here */
+Pet.prototype.vocalize = function() { return this.vocalization; };
+// !!! PLEASE READ: After talking with Ben, we believe that the tests may not
+//                  be entirely correct. The tests don't test, for example,
+//                  if the object literal in Question 2 has a numberOfPets
+//                  attribute. It also fails on things that should (to the best
+//                  of my ability to read the spec) pass. Also, no where in
+//                  any of the questions relating to pet or Pet does it say that
+//                  the vocalize function should return a string- and if we are
+//                  making a vocalize function it would make sense to have it
+//                  return 'vocalization' (which I took to be the string).
+//                  E.g. const doggy = new Pet('dog', 'wolf', 'hound', 'Bark!');
+//
+//                  If I am incorrect on any of the following please let me know
+//                  as it is quite possible I am missing something.
 
```

Comment:
Tests were not expanded to include prototypes as #139 suggests. Don't open an issue, as there already is one.